src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,184 @@
+set motor to lower lift 
+shift into low or high gear 
+read top limit switch 
+read bottom limit switch 
+read other limit switches...? 
+ go to specified height
+
+Commands 
+go to specified height 
+make the robot climb 
+raise lift 
+lower lift 
+*/
+
+package frc.team166.robot.subsystems;
+
+import javax.lang.model.util.ElementScanner6;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import frc.team166.chopshoplib.commands.ActionCommand;
+import frc.team166.chopshoplib.commands.CommandChain;
+import frc.team166.chopshoplib.commands.SubsystemCommand;
+import frc.team166.robot.RobotMap;
+import frc.team166.robot.RobotMap.Encoders;
+import edu.wpi.first.wpilibj.DigitalInput;
+
+/** 
+ * An example subsystem.  You can replace me with your own Subsystem. 
+ */
+public class Lift extends Subsystem {
+    // Defines Limit Switches (Digital imputs)
+    DigitalInput TopLimitSwitchLift = new DigitalInput(RobotMap.DigitalImputs.LIMIT_SWITCH_LIFT_BOTTOM);
+    DigitalInput BottomLimitSwitchLift = new DigitalInput(RobotMap.DigitalImputs.LIMIT_SWITCH_LIFT_TOP);
+    // Defines Encoders 
+    Encoder liftEncoder = new Encoder(RobotMap.Encoders.LIFT_A, RobotMap.Encoders.LIFT_B);
+    // Defines Motors 
+    WPI_TalonSRX m_liftmotor = new WPI_TalonSRX(RobotMap.CAN.LIFT_MOTOR_A);
+    WPI_TalonSRX m_liftmotor2 = new WPI_TalonSRX(RobotMap.CAN.LIFT_MOTOR_B);
+    // Defines the previus motors as one motor 
+    SpeedControllerGroup liftdrive = new SpeedControllerGroup(m_liftmotor, m_liftmotor2);
+    DoubleSolenoid LiftTransmission = new DoubleSolenoid(RobotMap.Solenoids.LiftTransmission,
+            RobotMap.Solenoids.LiftTransmission2);
+
+    public enum LiftHeight {
+        //will be changed
+        kFloor(0), kSwitch(1), kPortal(2), kIntake(3), kScaleLow(4), kScaleHigh(5), kClimb(6);
+
+        @SuppressWarnings("MemberName")
+        private double value;
+
+        LiftHeight(double value) {
+            this.value = value;
+        }
+
+        public double get() {
+            return value;
+        }
+    }
+
+    public Lift() {
+        addChild(liftEncoder);
+        //will be changed
+        liftEncoder.setDistancePerPulse(.1);
+    }
+
+    private Boolean TopLimitSwitchOn() {
+        if (TopLimitSwitchLift.get())
+            return true;
+        else
+            return false;
+    }
+
+    private Boolean BottomLimitSwitchOn() {
+        if (BottomLimitSwitchLift.get())
+            return true;
+        else
+            return false;
+    }
+
+    //gets the Lift Encoder's distance
+    public double EncoderDistance = liftEncoder.getDistance();
+    //gets the Lift Encoder's speed
+    public double EncoderSpeed = liftEncoder.getRate();
```

Comment:
Fields belong at the top of the class. Also note that this will only happen once. You should get the distance in a command.

Reply:
They should be private

Reply:
Agree with Ben here, there isn't much reason to assign `liftEncoder.getDistance()` to a variable when you aren't modifying it in any way. Wherever `EncoderDistance` is used later on could be replaced directly with `liftEncoder.getDistance()`